### PR TITLE
obtaining-api-versions-automatically

### PR DESCRIPTION
**Pull request by zikalino**

### CODE REVIEW

File: library/azure_rm_resource.py [hunk 1]
```diff
@@ -260,6 +258,19 @@ def exec_module(self, **kwargs):
 
             if orphan is not None:
                 self.url += '/' + orphan
+
+        # if api_version was not specified, get latest one
+        if not self.api_version:
+            # extract provider and resource type
+            provider = self.url.split("/providers/")[1].split("/")[0]
+            resourceType = self.url.split(provider + "/")[1].split("/")[0]
```
Comment (yungezz) [inline library/azure_rm_resource.py]: what if it's url has subresource like:
/subscriptions/xxx/resourceGroups/api2018/providers/Microsoft.Network/networkInterfaces/acctni
does  it  work?

Comment (zikalino) [reply library/azure_rm_resource.py]: yes

File: library/azure_rm_resource.py [hunk 2]
```diff
@@ -260,6 +258,19 @@ def exec_module(self, **kwargs):
...
+            url = "/subscriptions/" + self.subscription_id + "/providers/" + provider
+            api_versions = json.loads(self.mgmt_client.query(url, "GET", { 'api-version': '2015-01-01'}, None, None, [200], 0, 0).text)
+            for rt in api_versions['resourceTypes']:
+                if rt['resourceType'].lower() == resourceType.lower():
```
Comment (yungezz) [inline library/azure_rm_resource.py]: error message on not matching

Comment (zikalino) [reply library/azure_rm_resource.py]: ok, I will fail if api_version not found

File: library/azure_rm_resource.py [hunk 3]
```diff
@@ -260,6 +258,19 @@ def exec_module(self, **kwargs):
 
             if orphan is not None:
                 self.url += '/' + orphan
+
+        # if api_version was not specified, get latest one
+        if not self.api_version:
+            # extract provider and resource type
+            provider = self.url.split("/providers/")[1].split("/")[0]
```
[the remark this answers is not in the document]
Comment (zikalino) [reply library/azure_rm_resource.py]: i think we may have try and self.fail

[the remark this answers is not in the document]
Comment (yungezz) [reply library/azure_rm_resource.py]: i mean url.split(xx)[1] what if split return original string when url  is not splittable by /provider

File: library/azure_rm_resource.py (hunk 1 above)
Comment (zikalino) [reply library/azure_rm_resource.py]: if there's no "providers/xx/xx" then we can't get api version, so it should fail

**Pull request merged**